src/target/drivers/filesystems/syscalls.c

```diff
@@ -33,49 +33,47 @@ extern void printstr(char *, int);
 #define dbgMsec()          ((DEBUG_SYSCALLS) ? CLOCK_getms() : 0)
 
 static DIR   dir;
+
 #ifdef MEDIA_DRIVE
-static FATFS fat[2];
+static DRIVE drive[2];
 #else
-static FATFS fat[1];
+static DRIVE drive[1];
 #endif
 
-#ifdef FIL  // If FIL is #defined, it maps to FATFS
-    #define fil fat
-#else
-    static FIL fil[1];
-#endif
 extern u8 _drive_num;
 
 extern void init_err_handler();
 
 int FS_Mount();
 void FS_Unmount();
 
-intptr_t _open_r(FIL *r, const char *file, int flags, int mode);
-int _close_r(FIL *r);
+intptr_t _open_r(FSHANDLE *r, const char *file, int flags, int mode);
+int _close_r(FSHANDLE *r);
 
 
-int _read_r(FIL *r, char * ptr, int len);
-int _write_r(FIL *r, char * ptr, int len);
-int _lseek_r(FIL *r, int ptr, int dir);
+int _read_r(FSHANDLE *r, char * ptr, int len);
+int _write_r(FSHANDLE *r, char * ptr, int len);
+int _lseek_r(FSHANDLE *r, int ptr, int dir);
 
 
-int FS_Mount(void *_f, const char *drive)
+int FS_Init()
 {
-    (void)drive;
-    if(! _f) {
 #ifdef MEDIA_DRIVE
-        int res = (FS_Mount(&fat[0], "")  && FS_Mount(&fat[1], "media"));
+    int res = (FS_Mount(&drive[0].fat, "")  && FS_Mount(&drive[1].fat, "media"));
```

Comment:
I would rather to define "media" as a const, it is referenced so many times in this file.

Reply:
See above.  no reason to clean this further as the purpose is to get rid of it altogether